src/Aks/Aks/Commands/CreateOrUpdateKubeBase.cs

```diff
@@ -451,5 +457,47 @@ protected string DefaultDnsPrefix()
             var subPart = string.Join("", DefaultContext.Subscription.Id.Take(4));
             return $"{namePart}{subPart}";
         }
+
+        protected string GenerateSshKeyValue()
+        {
+            String generateSshKeyPath = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.UserProfile), ".ssh", "id_rsa"); ;
+            if (File.Exists(generateSshKeyPath))
+            {
+                throw new ArgumentException(string.Format(Resources.DefaultSshKeyAlreadyExist));
+            }
+            using (Process process = new Process())
+            {
+                process.StartInfo.FileName = "ssh-keygen.exe";
```

Comment:
Could we generate SSH key using C# code instead of calling ssh-keygen.exe considering ssh-keygen.exe is available only on Windows?